hamiltonian_flow/models/harmonic_oscillator.py

```diff
@@ -0,0 +1,256 @@
+from functools import cached_property
+from typing import Dict, Literal, Optional, Union
+
+import numpy as np
+import pandas as pd
+from pydantic import BaseModel, computed_field, field_validator
+
+
+class HarmonicOscillatorSystem(BaseModel):
```

Comment:
Is a user supposed to use this class, or is it intended to be private? This question also applies to `HarmonicOscillatorIC`.

Reply:
https://github.com/kausalflow/hamiltonian_flow/pull/8#discussion_r1501906946

Yes, that is the plan. Once we have a few models at hand, we can create a yaml file as a set of standard params. This will become our standard dataset. However, this is for later.

Reply:
> Is a user supposed to use this class, or is it intended to be private? This question also applies to `HarmonicOscillatorIC`.

Good question. I am not sure. Let's see. We can refactor it to be weekly private if the users don't need it.